### PR TITLE
Wrap splits functions - for ParMmg convenience

### PR DESCRIPTION
**Pull request by laetitia-m**
1. **Wrap the splits functions** used in LS mode (split1, split2sf, split3cone and split4op) for ParMmg convenience. In the splits, the configuration is determined based on the vertices indices. In ParMmg, we need to use the global vertices numbering instead of the local (in term of proc) vertices numbering: to allow this, split functions have been modified to be able to specify the tetra vertices numbering as argument. This change has no impact on mmg.
2. **Rename** some **split configuration functions** for consistency (e.g. `MMG3D_configSplit3op` becomes `MMG3D_split3op_cfg`).

### CODE REVIEW

**Review by Algiane — COMMENTED** [review]
Hi,

I have few remarks on the factorization choice. It can be discussed tomorrow if needed.

Best

File: src/common/hash.c [hunk 1]
```diff
@@ -424,6 +424,47 @@ int MMG5_hashUpdate(MMG5_Hash *hash, MMG5_int a,MMG5_int b,MMG5_int k) {
...
+
+    if ( !ph->nxt ) return 0;
+
+    ph = &hash->item[ph->nxt];
+
+  }
+
+  return 0;
+}
+
```
[the remark this answers is not in the document]
Comment (laetitia-m) [reply src/common/hash.c]: `MMG5_hashUpdate_all` has been removed from mmg and replaced by `PMMG_hashUpdate_all` in ParMmg.

File: src/mmg3d/split_3d.c [hunk 2]
```diff
@@ -128,12 +128,32 @@ int MMG3D_split1_sim(MMG5_pMesh mesh,MMG5_pSol met,MMG5_int k,MMG5_int vx[6]) {
...
+ * \param vGlobNum vertices indices of the tetra k.
+ * \param metRidTyp metric storage (classic or special)
+ *
  * \return 0 if fail, 1 otherwise
  *
  * Split 1 edge of tetra \a k.
  *
  */
-int MMG5_split1(MMG5_pMesh mesh,MMG5_pSol met,MMG5_int k,MMG5_int vx[6],int8_t metRidTyp) {
+int MMG5_split1_globNum(MMG5_pMesh mesh,MMG5_pSol met,MMG5_int k,MMG5_int vx[6],MMG5_int vGlobNum[4],int8_t metRidTyp) {
```
[the remark this answers is not in the document]
Comment (laetitia-m) [reply src/mmg3d/split_3d.c]: Ok, `MMG5_split1` has been kept as it was.

**Comment by Algiane** [comment]
Thanks

**Pull request merged**